src/test/java/org/broadinstitute/hellbender/tools/spark/pipelines/MarkDuplicatesSparkIntegrationTest.java

```diff
@@ -175,6 +174,308 @@ public void testMarkDuplicatesSparkIntegrationTestLocal(
+
+        ArgumentsBuilder args = new ArgumentsBuilder();
+        args.add("--" + StandardArgumentDefinitions.INPUT_LONG_NAME);
+        args.add(input.getPath());
+        args.add("--" + StandardArgumentDefinitions.OUTPUT_LONG_NAME);
+
+        File outputFile = createTempFile("markdups", ".bam");
+        outputFile.delete();
+        args.add(outputFile.getAbsolutePath());
+
+        args.add("--" + StandardArgumentDefinitions.METRICS_FILE_LONG_NAME);
+        File metricsFile = createTempFile("markdups_metrics", ".txt");
+        args.add(metricsFile.getAbsolutePath());
+
+        args.add("--" + MarkDuplicatesSparkArgumentCollection.REMOVE_SEQUENCING_DUPLICATE_READS);
+
+        runCommandLine(args.getArgsArray());
+
+        Assert.assertTrue(outputFile.exists(), "Can't find expected MarkDuplicates output file at " + outputFile.getAbsolutePath());
+
+        int totalReads = 0;
+        int duplicateReads = 0;
+        try (final ReadsDataSource outputReads = new ReadsDataSource(outputFile.toPath())) {
+            for (GATKRead read : outputReads) {
+                ++totalReads;
+
+                if (read.isDuplicate()) {
+                    ++duplicateReads;
+                }
+            }
+        }
+
+        final MetricsFile<GATKDuplicationMetrics, Comparable<?>> metricsOutput = new MetricsFile<>();
+        try {
+            metricsOutput.read(new FileReader(metricsFile));
+        } catch (final FileNotFoundException ex) {
+            System.err.println("Metrics file not found: " + ex);
+        }
+        final List<GATKDuplicationMetrics> nonEmptyMetrics = metricsOutput.getMetrics().stream().filter(
+                metric ->
+                        metric.UNPAIRED_READS_EXAMINED != 0L ||
+                                metric.READ_PAIRS_EXAMINED != 0L ||
+                                metric.UNMAPPED_READS != 0L ||
+                                metric.UNPAIRED_READ_DUPLICATES != 0L ||
+                                metric.READ_PAIR_DUPLICATES != 0L ||
+                                metric.READ_PAIR_OPTICAL_DUPLICATES != 0L ||
+                                (metric.PERCENT_DUPLICATION != null && metric.PERCENT_DUPLICATION != 0.0 && !Double.isNaN(metric.PERCENT_DUPLICATION)) ||
+                                (metric.ESTIMATED_LIBRARY_SIZE != null && metric.ESTIMATED_LIBRARY_SIZE != 0L)
+        ).collect(Collectors.toList());
+
+        // Asserting
+        int expectedOpticalDuplicatesGroups = 0;
+        for (int i = 0; i < nonEmptyMetrics.size(); i++ ){
+            final GATKDuplicationMetrics observedMetrics = nonEmptyMetrics.get(i);
+            List<?> expectedList = metricsExpected.get(observedMetrics.LIBRARY);
+            expectedOpticalDuplicatesGroups += (Long) expectedList.get(5);
+        }
+        // NOTE: this test will fail if we add a more comprehensive example set with optical duplicates containing secondary/supplementary reads
```

Comment:
this makes it sound like there's a bug in the code, maybe make it clear that that it's the test that can't handle a more complicated scenerio?